polaris-react/src/components/Badge/components/Pip.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+import {classNames, variationName} from '../../../utilities/css';
+import styles from '../Badge.scss';
+import type {Progress, Status} from '../types';
+
+export interface PipProps {
+  children?: '' | JSX.Element | undefined;
```

Comment:
I think you should use
```suggestion
  children?: React.ReactNode;
```

Although, do we really want to allow `children`? The `Pip` is inline so consumers can just do:
```jsx
<p>
  <Pip />
  order #123
</p>
```

Just like in the badge, no?

---

However, since you didn’t port over any of the built-in accessibility markup stuff from the `Badge` (nor the override prop), and because `children` is optional, right now you risk having consumers render a `Pip` without any associated meaning.

IMO we should port all of the status + progress logic from the `Badge` into `Pip`.

Reply:
Great review @Tom-Bonnike 🙏

In `Badge` we had before:
```` tsx

const accessibilityLabel = statusAndProgressLabelOverride
    ? statusAndProgressLabelOverride
    : i18n.translate('Polaris.Badge.progressAndStatus', {
        progressLabel,
        statusLabel,
      });

 ...

  let accessibilityMarkup = hasAccessibilityLabel && (
    <VisuallyHidden>{accessibilityLabel}</VisuallyHidden>
  );

  if (progressLabel && !icon) {
    accessibilityMarkup = (
      <span className={styles.Pip}>{accessibilityMarkup}</span>
    );
  }
````

Wouldn't `<span aria-label={accessibilityLabel} className={styles.Pip} />` be enough from an accessibility perspective?

If so,  we could simplify `Pip` interface to something like bellow, couldn't we?
````tsx
export interface PipProps {
  status?: Status;
  progress?: Progress;
  accessibilityLabel: string;
}
````

Reply:
I’m not from the Polaris team and only recently touched the `Badge` hence why my name was suggested for review, so take this with a grain of salt 😛 

---

I was also talking about the labels, I got it wrong though we only render the Pip if there is the `progress` prop.
```jsx
  let progressLabel = '';
  switch (progress) {
    case PROGRESS_LABELS.incomplete:
      progressLabel = i18n.translate(
        'Polaris.Badge.PROGRESS_LABELS.incomplete',
      );
      break;
    case PROGRESS_LABELS.partiallyComplete:
      progressLabel = i18n.translate(
        'Polaris.Badge.PROGRESS_LABELS.partiallyComplete',
      );
      break;
    case PROGRESS_LABELS.complete:
      progressLabel = i18n.translate('Polaris.Badge.PROGRESS_LABELS.complete');
      break;
  }

  let statusLabel = '';
  switch (status) {
    case STATUS_LABELS.info:
      statusLabel = i18n.translate('Polaris.Badge.STATUS_LABELS.info');
      break;
    case STATUS_LABELS.success:
      statusLabel = i18n.translate('Polaris.Badge.STATUS_LABELS.success');
      break;
    case STATUS_LABELS.warning:
      statusLabel = i18n.translate('Polaris.Badge.STATUS_LABELS.warning');
      break;
    case STATUS_LABELS.critical:
      statusLabel = i18n.translate('Polaris.Badge.STATUS_LABELS.critical');
      break;
    case STATUS_LABELS.attention:
      statusLabel = i18n.translate('Polaris.Badge.STATUS_LABELS.attention');
      break;
    case STATUS_LABELS.new:
      statusLabel = i18n.translate('Polaris.Badge.STATUS_LABELS.new');
      break;
  }

  const accessibilityLabel = statusAndProgressLabelOverride
    ? statusAndProgressLabelOverride
    : i18n.translate('Polaris.Badge.progressAndStatus', {
        progressLabel,
        statusLabel,
      });
  const hasAccessibilityLabel =
    progressLabel || statusLabel || statusAndProgressLabelOverride;
  let accessibilityMarkup = hasAccessibilityLabel && (
    <VisuallyHidden>{accessibilityLabel}</VisuallyHidden>
  );

  if (progressLabel && !icon) {
    accessibilityMarkup = (
      <Pip progress={progress} status={status}>
        {accessibilityMarkup}
      </Pip>
    );
  }

  return (
    <span className={className}>
      {accessibilityMarkup}
      {icon && (
        <span className={styles.Icon}>
          <Icon source={icon} />
        </span>
      )}
      {children && <span>{children}</span>}
    </span>
  );
}
```

> If so, we could simplify Pip interface to something like bellow, couldn't we?

Yeah, but I think the label should be an override though.

Reply:
👋 

> Wouldn't <span aria-label={accessibilityLabel} className={styles.Pip} /> be enough from an accessibility perspective?

Some assistive tech will ignore an `aria-label` added to an elements without semantic meaning. For this reason, we avoid adding `aria-label` to a span, which is why we pass the a11y content as a child